tests/test_prometheus_alerts_configmap.py

```diff
@@ -1,50 +1,102 @@
 import yaml
 from tests.helm_template_generator import render_chart
+from . import supported_k8s_versions
+import re
+import pytest
 
 
-def process_record(rule):
-    """Process a record rule.
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+class TestPrometheusAlertConfigmap:
+    show_only = ["charts/prometheus/templates/prometheus-alerts-configmap.yaml"]
 
-    https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/
-    """
-    assert isinstance(rule.get("expr"), str)
-    assert isinstance(rule.get("record"), str)
+    @staticmethod
+    def process_record(rule):
+        """Process a record rule.
 
+        https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/
+        """
+        assert isinstance(rule.get("expr"), str)
+        assert isinstance(rule.get("record"), str)
 
-def process_alert(rule):
-    """Process an alert rule.
+    @staticmethod
+    def process_alert(rule):
+        """Process an alert rule.
 
-    https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/
-    """
-    assert isinstance(rule.get("alert"), str)
-    assert isinstance(rule.get("expr"), str)
-    if rule["alert"] != "Watchdog":
-        assert isinstance(rule.get("for"), str)
-    assert isinstance(rule.get("labels"), dict)
-    assert isinstance(rule.get("annotations"), dict)
+        https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/
+        """
+        assert isinstance(rule.get("alert"), str)
+        assert isinstance(rule.get("expr"), str)
+        if rule["alert"] != "Watchdog":
```

Comment:
```
else:
    assert "alert" not in rule
```
?

Reply:
i think we could do something even cooler.
```
assert (rule["alert"] == "Watchdog") == ("for" not in rule)
```

Reply:
I like the first suggestion, but I think the second one would confuse people. Going with the first one.